pkg/apis/rollouts/validation/validation_references_test.go

```diff
@@ -472,11 +514,25 @@ func TestValidateVirtualService(t *testing.T) {
 		assert.Equal(t, expectedErr.Error(), allErrs[0].Error())
 	})
 
+	t.Run("validate virtualService TCP routes - success", func(t *testing.T) {
+		vsvc := unstructured.StrToUnstructuredUnsafe(successCaseTcpVsvc)
+		allErrs := ValidateVirtualService(ro, *vsvc)
+		assert.Empty(t, allErrs)
+	})
+
+	t.Run("validate virtualService TCP routes - failure", func(t *testing.T) {
```

Comment:
Same as above. This test is not really testing any new code that you are checking in.